src/tree/SubscriptionTreeItem.ts

```diff
@@ -102,8 +104,15 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
         this._treeMap = {};
         const groupBySetting = <string>settingUtils.getGlobalSetting<string>('groupBy');
 
-        for (const rgTree of this._items) {
-            (<AppResourceTreeItem>rgTree).mapSubGroupConfigTree(context, groupBySetting);
+        const client: ResourceManagementClient = await createResourceClient([context, this]);
+        const resourceGroups = uiUtils.listAllIterator(client.resourceGroups.list());
```

Comment:
I'm only about 30% sure, but I think this is preemptively calling `client.resourceGroups.list()` (before the `await resourceGroups` below), and that this would make it not do that:
```suggestion
        const resourceGroups = uiUtils.listAllIterator(() => client.resourceGroups.list());
```

Reply:
Wouldn't `uiUtils.listAllIterator(() => client.resourceGroups.list());` result in `client.resourceGroups.list()` being called every time the `resourceGroups` Promise is awaited?

Also I think we are ok with it calling `client.resourceGroups.list()` before the await, since we don't mind when it starts/finishes.

Reply:
If we're starting it no matter what, then it's better as-written, because it will start the `client.resourceGroups.list()` immediately, rather than waiting until `await resourceGroups`.